resource_dispatcher/resources/secrets/.gitignore

```diff
@@ -1,4 +1,4 @@
 # Ignore everything in this directory
 *
 # Except this file
-!.gitignore
\ No newline at end of file
+!.gitignore
```

Comment:
are these just to make sure the empty directory exists and doesn't get culled by git?

Reply:
If these directories are required to be baked into the image, then how are they also [configurable here](https://github.com/canonical/resource-dispatcher-image/blob/ef1db240eb91a4d78aa9c10ba6fb5c331005394a/resource_dispatcher/main.py#L40)?

I think a typical pattern would be to:
* have the directory [configurable](https://github.com/canonical/resource-dispatcher-image/blob/ef1db240eb91a4d78aa9c10ba6fb5c331005394a/resource_dispatcher/main.py#L40) like you do
* not bake these directories into the charm
* expect the user to copy files into an existing location, mount a volume, or create the location as needed

In your case, probably creating the location is appropriate.  What you could do remove these directories, then update [this `self.container.pus(...)`](https://github.com/canonical/resource-dispatcher/blob/1185adfc490973fd88008278a5d50cae3d88d9ef/src/charm.py#L219) to be `self.container.push(..., make_dirs=True)`

Reply:
imo this directory structure is the choice of the user.  In this script, you recursively glob everything and don't actually rely on the directory structure, so I don't think it makes sense to bake a directory structure in the image.  

I do understand why you want the directory structure though (it makes it easier to organise the things the charm creates).  That makes lots of sense - but if it is the charm that requires it and does it, then we shouldn't implement half of it down in this image.  The charm can make its own directories 

Let's leave this for now, but take an action later to modify the charm.  the easiest way I can think of is to have the `_sync_manifests` function in the charm create the directory structure it wants before doing any syncing.